src/topology/uniform_space/equicontinuity.lean

```diff
@@ -0,0 +1,341 @@
+/-
+Copyright (c) 2022 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import topology.uniform_space.uniform_convergence_topology
+
+/-!
+# Equicontinuity of a family of functions
+
+## Main definitions
+
+* `equicontinuous_at`: equicontinuity of a family of functions at a point
+* `equicontinuous`: equicontinuity of a family of functions on the whole domain
+* `uniform_equicontinuous`: uniform equicontinuity of a family of functions on the whole domain
```

Comment:
Very often, we have started with purely local and global notions, like `continuous_at` and `continuous`, only to discover later on that relative versions `continuous_within_at` and `continuous_on` were relevant. That's why we have also relative versions of differentiability, smoothness, and so on (and all these turn out to be very important in applications). My guess is that the same would apply here -- but if you don't want to do it in this PR I'll understand perfectly well.